shared/AppInsightsCommon/src/Offline.ts

```diff
@@ -58,26 +93,58 @@ export function createOfflineListener(parentEvtNamespace?: string | string[]): I
                 eventOn(target, "offline", _setOffline, _evtNamespace);
             }
         }
-
         return enabled;
     }
 
+    function _isOnline(){
+        return _onlineStatus;
+    }
+
+    function currentState(){
+        if (uState === eOfflineValue.Offline || rState === eOfflineValue.Offline){
+            return false;
+        }
+        return true; // if both unknown, then we assume the network is good
+    }
+
+    function listnerNoticeCheck(){
+        // we were offline and are now online or we were online and now offline
+        if (_isOnline() !== currentState()) {
+            _onlineStatus = currentState(); // use the resolved state to update
+            // send all the callbacks with the current state
+            listenerList.forEach((callback: OfflineCallback) => {
```

Comment:
Please use `arrForEach` as we have had issue reported previously when we used it as some 3rd party library included some additional properties to the "array" prototype which broken this pattern.